staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -64,6 +64,8 @@ func (w *realTimeoutFactory) TimeoutCh() (<-chan time.Time, func() bool) {
 // serveWatch will serve a watch response.
 // TODO: the functionality in this method and in WatchServer.Serve is not cleanly decoupled.
 func serveWatch(watcher watch.Interface, scope *RequestScope, mediaTypeOptions negotiation.MediaTypeOptions, req *http.Request, w http.ResponseWriter, timeout time.Duration) {
+	defer watcher.Stop()
```

Comment:
Hi @answer1991 , why do we need this when we already have the watcher to be closed after timeout in `cacheWatcher.process()`.